pandas/tests/io/json/test_normalize.py

```diff
@@ -375,3 +375,32 @@ def test_nonetype_dropping(self):
              'info.last_updated': '26/05/2012'}]
 
         assert result == expected
+
+    def test_nonetype_multiple_levels(self):
+        # GH21158: If inner level json has a key with a null value
+        # make sure it doesnt do a new_d.pop twice and except
+        data = {
+            "id": None,
```

Comment:
does this have the same result if id is NOT repeated (with None), just missing at various levels? e.g. try with a single id with None at the top and bottom levels (in another case, leave this one as well)